contrib/meson/meson/programs/meson.build

```diff
@@ -7,46 +7,54 @@
 # in the COPYING file in the root directory of this source tree).
 # #############################################################################
 
-lz4_root_dir = '../../../..'
-
-lz4_includes = include_directories(join_paths(lz4_root_dir, 'programs'))
-lz4_sources = [join_paths(lz4_root_dir, 'programs/bench.c'),
-  join_paths(lz4_root_dir, 'programs/datagen.c'),
-  join_paths(lz4_root_dir, 'programs/lz4cli.c'),
-  join_paths(lz4_root_dir, 'programs/lz4io.c')]
-lz4_c_args = []
-
-export_dynamic_on_windows = false
-# explicit backtrace enable/disable for Linux & Darwin
-if not use_backtrace
-  lz4_c_args += '-DBACKTRACE_ENABLE=0'
-elif use_debug and host_machine_os == os_windows  # MinGW target
-  lz4_c_args += '-DBACKTRACE_ENABLE=1'
-  export_dynamic_on_windows = true
+sources = files(
+  lz4_source_root / 'programs/bench.c',
+  lz4_source_root / 'programs/datagen.c',
+  lz4_source_root / 'programs/lz4cli.c',
+  lz4_source_root / 'programs/lz4io.c',
+)
+
+lz4 = executable(
+  'lz4',
+  sources,
+  objects: liblz4.extract_all_objects(recursive: true),
+  include_directories: include_directories(lz4_source_root / 'programs'),
+  dependencies: [liblz4_dep],
+  export_dynamic: get_option('debug') and host_machine.system() == 'windows',
+  install: true
+)
+
+install_man(lz4_source_root / 'programs/lz4.1')
+
+if meson.version().version_compare('>=0.61.0')
+  install_symlink(
```

Comment:
@eli-schwartz could you review this usage of install_symlink?

Reply:
`pointing_to` needs to be a string, not a meson `executable()` object.

So, since you want to create `lz4c` as a program which is a symlink to the `lz4` program, just use `pointing_to: 'lz4'`. (Meson then creates a relative symlink that tries to find the lz4 destination in the same directory as the symlink itself.)

Reply:
I'd use a foreach loop. `foreach alias: ['lz4c', 'lz4cat', 'unlz4']`, create two symlinks, one in the bindir pointing to `'lz4'`, one in the mandir named `alias + '.1'` pointing to `'lz4.1'`

Reply:
It seems a bit unusual to only create the aliases when the user has a new enough version of Meson. Though there is no denying it's a lot cleaner to use the Meson function.

Reply:
Since this is a community contributed port and the symlinks are deprecated as far as I can tell, I just wanted to say screw it and use the Meson functions.

Reply:
Yeah, it's not super important for them to exist I guess so having it be less well supported is probably fine.